src/superqt/_vendor/dataframeeditor/dataframeeditor.py

```diff
@@ -0,0 +1,1302 @@
+        self.create_table_index()
+
+        # Create the model and view of the data
+        self.dataModel = DataFrameModel(data, parent=self)
+        self.dataModel.dataChanged.connect(self.save_and_close_enable)
+        self.create_data_table()
+
+        self.layout.addWidget(self.hscroll, 2, 0, 1, 2)
+        self.layout.addWidget(self.vscroll, 0, 2, 2, 1)
+
+        # autosize columns on-demand
+        self._autosized_cols = set()
+        # Set limit time to calculate column sizeHint to 300ms,
+        # See spyder-ide/spyder#11060
+        self._max_autosize_ms = 300
+        self.dataTable.installEventFilter(self)
+
+        avg_width = self.fontMetrics().averageCharWidth()
+        self.min_trunc = avg_width * 12  # Minimum size for columns
+        self.max_width = avg_width * 64  # Maximum size for columns
+
+        self.setLayout(self.layout)
+        # Make the dialog act as a window
+        self.setWindowFlags(Qt.Window)
+        btn_layout = QHBoxLayout()
+        btn_layout.setSpacing(5)
+
+        btn_format = QPushButton(trans_("Format"))
+        # disable format button for int type
+        btn_layout.addWidget(btn_format)
+        btn_format.clicked.connect(self.change_format)
+
+        btn_resize = QPushButton(trans_("Resize"))
+        btn_layout.addWidget(btn_resize)
+        btn_resize.clicked.connect(self.resize_to_contents)
+
+        bgcolor = QCheckBox(trans_("Background color"))
+        bgcolor.setChecked(self.dataModel.bgcolor_enabled)
+        bgcolor.setEnabled(self.dataModel.bgcolor_enabled)
+        bgcolor.stateChanged.connect(self.change_bgcolor_enable)
+        btn_layout.addWidget(bgcolor)
+
+        self.bgcolor_global = QCheckBox(trans_("Column min/max"))
+        self.bgcolor_global.setChecked(self.dataModel.colum_avg_enabled)
+        self.bgcolor_global.setEnabled(
+            not self.is_series and self.dataModel.bgcolor_enabled
+        )
+        self.bgcolor_global.stateChanged.connect(self.dataModel.colum_avg)
+        btn_layout.addWidget(self.bgcolor_global)
+
+        btn_layout.addStretch()
+
+        self.btn_save_and_close = QPushButton(trans_("Save and Close"))
+        self.btn_save_and_close.setDisabled(True)
+        self.btn_save_and_close.clicked.connect(self.accept)
+        btn_layout.addWidget(self.btn_save_and_close)
+
+        self.btn_close = QPushButton(trans_("Close"))
+        self.btn_close.setAutoDefault(True)
+        self.btn_close.setDefault(True)
+        self.btn_close.clicked.connect(self.reject)
+        btn_layout.addWidget(self.btn_close)
+
+        btn_layout.setContentsMargins(0, 16, 0, 16)
+        self.layout.addLayout(btn_layout, 4, 0, 1, 2)
+        self.setModel(self.dataModel)
+        self.resizeColumnsToContents()
+
+        return True
+
+    def accept(self, e):
+        self.dataModel.df.to_pickle("tmp.pkl")
+        self.close()
+
+    @Slot(QModelIndex, QModelIndex)
+    def save_and_close_enable(self, top_left, bottom_right):
```

Comment:
this and probably most of the other methods should be private, we can discuss exactly what the public API should be